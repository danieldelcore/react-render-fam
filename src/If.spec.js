import React from 'react';
import renderer from 'react-test-renderer';
import { If } from '.';

test('Renders the content if predicate is true', () => {
    const value = true;
    const component = renderer.create(
        <If predicate={value}>
            <p>Hello!</p>
        </If>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('Renders the content if predicate supplied as a boolean attribute', () => {
    const component = renderer.create(
        <If predicate>
            <p>Hello!</p>
        </If>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('Omits the content if predicate not supplied', () => {
    const component = renderer.create(
        <If>
            <p>Hello!</p>
        </If>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('Omits the content if predicate is false', () => {
    const value = false;
    const component = renderer.create(
        <If predicate={value}>
            <p>Hello!</p>
        </If>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});
