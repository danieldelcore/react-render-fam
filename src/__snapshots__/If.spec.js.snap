// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Omits the content if predicate is false 1`] = `""`;

exports[`Omits the content if predicate not supplied 1`] = `""`;

exports[`Renders the content if predicate is true 1`] = `
<p>
  Hello!
</p>
`;

exports[`Renders the content if predicate supplied as a boolean attribute 1`] = `
<p>
  Hello!
</p>
`;
